### PR TITLE
priority queue

### PR DESCRIPTION
**Pull request by aepaysinger**
my only question is for the pop method. is it okay that it returns the value (priority, value) or does it need to return just the values value?

### CODE REVIEW

File: binheap.py [hunk 1]
```diff
@@ -74,7 +74,7 @@ def _heap_down_pop(self):
                     else:
                         return
                 else:
-                    if self._left_child_value > self._storage[index]:
+                    if self._left_child_value(index) > self._storage[index]:
```
Comment (W-Ely) [inline binheap.py:73]: If this was a bug that was found, write the test that would have failed before the change.

File: priority_que.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
+import bisect
```
Comment (W-Ely) [inline priority_que.py]: No longer need bisect.

File: priority_que.py [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    def __len__(self):
+        return self._storage._size
+
+    def insert(self, value, priority=0):
+        self._storage.push(Node(priority, value))
+
+    def pop(self):
+        try:
+            return self._storage.pop().value
+        except IndexError:
```
Comment (W-Ely) [inline priority_que.py]: We also missed this in the BH review. It shouldn't raise an IndexError, and neither should the Queue.  Neither of them have the concept of an Index.

File: priority_que.py [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+            raise IndexError("No items to pop.")
+
+    def peek(self):
+        try:
+            return self._storage._storage[0].value
+        except IndexError:
+            raise IndexError("No items to see.")
```
Comment (W-Ely) [inline priority_que.py]: While catching the Index error here is Okay, Queues don't have Index, so throwing an Index Error isn't correct.

File: priority_que.py [hunk 5]
```diff
@@ -0,0 +1,41 @@
+import bisect
+
+from binheap import BinaryHeap
+
+
+class PriorityQue:
```
Comment (W-Ely) [inline priority_que.py]: Spell this out PriorityQueue

**Review by W-Ely — COMMENTED** [review]
Make sure you have a test that ensures that if the priorities are the same, the queue operates on a first in first out (fifo) basis.

File: priority_que.py (hunk 3 above)
Comment (aepaysinger) [reply priority_que.py]: i originally had it has ValueError but it failed. I didnt realize you can except one error and raise another

Comment (W-Ely) [reply priority_que.py]: Why are the tests passing even though you are catching an IndexError here? Once you figure it out be sure your have tests to cover the case.

**Review by W-Ely — APPROVED** [review]

**Pull request merged**